### PR TITLE
Move HttpUtility to System in unityjit.

### PR DESCRIPTION
**Pull request by joncham**
This change moves HttpUtility and a few helpers from System.Web to the System assembly. The reduces the dependencies of netstandard.dll greatly. This makes it easier for us to enable Mono players to always use the `unityjit` profile even with API compatibility level of ".NET Standard 2.0". The build size will increase slightly, but much less than without this PR. For example, we have a test today where netstandard.dll pulls in the following extra assemblies when targeting unityjit rather than unityaot

Trunk:
```
"assets/bin/Data/Managed/System.Design.dll",
"assets/bin/Data/Managed/System.Web.ApplicationServices.dll",
"assets/bin/Data/Managed/System.Windows.Forms.dll",
"assets/bin/Data/Managed/System.Drawing.Design.dll",
"assets/bin/Data/Managed/Mono.WebBrowser.dll",
"assets/bin/Data/Managed/Mono.Posix.dll",
"assets/bin/Data/Managed/System.Runtime.Serialization.Formatters.Soap.dll",
"assets/bin/Data/Managed/System.ComponentModel.DataAnnotations.dll",
"assets/bin/Data/Managed/System.Web.dll",
"assets/bin/Data/Managed/System.Web.Services.dll",
"assets/bin/Data/Managed/System.Security.dll",
"assets/bin/Data/Managed/Novell.Directory.Ldap.dll",
"assets/bin/Data/Managed/Accessibility.dll",
"assets/bin/Data/Managed/Mono.Data.Sqlite.dll",
"assets/bin/Data/Managed/System.DirectoryServices.dll",
```

With this PR:
```
"assets/bin/Data/Managed/Mono.Posix.dll",
"assets/bin/Data/Managed/System.Security.dll",
```

Which I think is acceptable.

A type forwarder is setup to not break compatibility. Note, there are two small changes in behavior. The defines in `HttpEncoder.cs` and `HttpUtility.cs` hardcode some things rather than using System.Configuration. I think this is okay as we already do this in the `unityaot` profile.

### CODE REVIEW

**Comment by joncham** [comment]
Note, I would do this same work for the upgrade branch if we think it makes sense to land this.

File: mcs/class/System.Web/System.Web.Util/HttpEncoder.cs [hunk 1]
```diff
@@ -30,12 +30,14 @@
 // OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 //
+#if !UNITY_FORWARD_TO_SYSTEM
```
Comment (UnityAlex) [inline mcs/class/System.Web/System.Web.Util/HttpEncoder.cs]: Just curious but would it be possible to just not include HttpEncoder.cs in the unityjit sources?

Comment (joncham) [reply mcs/class/System.Web/System.Web.Util/HttpEncoder.cs]: I tried for almost 2 hours to get the "*.excludes.sources" mechanism working but failed. Or we can copy/paste the entire source list and remove a few files.

Comment (UnityAlex) [reply mcs/class/System.Web/System.Web.Util/HttpEncoder.cs]: Using the exclude would be best but this is probably better than having yet another copy of all the sources just to exclude a file.

File: mcs/class/System.Web/System.Web.Util/HttpEncoder.cs [hunk 2]
```diff
@@ -30,12 +30,14 @@
...
+
 using System;
 using System.Collections.Generic;
 using System.Configuration;
 using System.IO;
 using System.Text;
-#if !NO_SYSTEM_WEB_DEPENDENCY && !MOBILE
+#if !NO_SYSTEM_WEB_DEPENDENCY && !MOBILE && !UNITY_JIT
```
Comment (UnityAlex) [inline mcs/class/System.Web/System.Web.Util/HttpEncoder.cs:39]: If `UNITY_FORWARD_TO_SYSTEM` is defined for the unityjit profile is this added check still needed?

**Review by joshpeterson — APPROVED** [review]

File: mcs/class/System.Web/System.Web.Util/HttpEncoder.cs (hunk 2 above)
Comment (UnityAlex) [reply mcs/class/System.Web/System.Web.Util/HttpEncoder.cs:39]: Nevermind since it's moved we still wind up in here so the check is required. :-)

**Review by UnityAlex — APPROVED** [review]

File: mcs/class/System.Web/System.Web.Util/HttpEncoder.cs (hunk 1 above)
Comment (joncham) [reply mcs/class/System.Web/System.Web.Util/HttpEncoder.cs]: @bholmes got the exclude file working for me ✋

**Pull request merged**